shpkpr/marathon/client.py

```diff
@@ -31,27 +31,25 @@ def _make_request(self, method, path, **kwargs):
         request = getattr(requests, method.lower())
         return request(self._build_url(path), **kwargs)
 
-    @property
-    def embed_params(self):
+    def embed_params(self, entity_type):
         return [
-            "app.tasks",
-            "app.counts",
-            "app.deployments",
-            "app.lastTaskFailure",
-            "app.taskStats",
+            "{0}.tasks".format(entity_type),
+            "{0}.counts".format(entity_type),
+            "{0}.deployments".format(entity_type),
+            "{0}.lastTaskFailure".format(entity_type),
+            "{0}.taskStats".format(entity_type),
         ]
 
     def get_application(self, application_id):
         """Returns detailed information for a single application.
         """
         path = "/v2/apps/" + application_id
-        params = {"embed": self.embed_params}
+        params = {"embed": self.embed_params("app")}
         response = self._make_request('GET', path, params=params)
 
         if response.status_code == 200:
             application = response.json()['app']
-            validate_app(application)
-            return application
+            return validate_app(application)
```

Comment:
Yeah, reading this makes me more convinced we should have the validator return a true / false and something else strip away the unneeded schema.

Also, where does `validate_app` come from? It looks like it's defined in another file? :confused: